src/Groups/group_constructors.jl

```diff
@@ -475,20 +475,26 @@ where the `i`-th generator is printed as `L[i]`.
 function free_group(n::Int, s::VarName = :f; eltype::Symbol = :letter)
    @req n >= 0 "n must be a non-negative integer"
    if eltype == :syllable
-     return FPGroup(GAP.Globals.FreeGroup(n, GAP.GapObj(s); FreeGroupFamilyType = GapObj("syllable"))::GapObj)
+     G = FPGroup(GAP.Globals.FreeGroup(n, GAP.GapObj(s); FreeGroupFamilyType = GapObj("syllable"))::GapObj)
    else
-     return FPGroup(GAP.Globals.FreeGroup(n, GAP.GapObj(s))::GapObj)
+     G = FPGroup(GAP.Globals.FreeGroup(n, GAP.GapObj(s))::GapObj)
    end
+   GAP.Globals.SetRankOfFreeGroup(G.X, n)
```

Comment:
I see: `RankOfFreeGroup` belongs to GAP's FGA package, therefore the function `FreeGroup` cannot set this attribute. Thus the situation here is better.

Reply:
Aye.

Regarding the definition of `rank(::GrpAbFinGen)` differing (it returns the Z-rank, or the "Hirsch length" to use polycyclic group terminology): ahh, that's a shame. Dang. Well, I could rename this function `_rank` for now, so that I can use it in the printing code (that was what motivated this PR), and then possibly again restrict it to support "free fp groups" only. And then worry about how to expose this to users later.